libraries/api-extractor-model/src/mixins/ApiExtendsMixin.ts

```diff
@@ -0,0 +1,393 @@
+        }
+      } else {
+        throw new InternalError('ApiExtendsMixin expects a base class that inherits from ApiDeclaredItem');
+      }
+    }
+
+    /**
+     * Finds all of the ApiItem's immediate and inherited members by walking up the inheritance tree.
+     *
+     * @param messageCallback - A callback that is called when this method hits some kind of error
+     * that could impact finding inherited members.
+     *
+     * @remarks
+     *
+     * For example, given the following class heritage:
+     *
+     * ```
+     * export class A {
+     *   a: number|boolean;
+     * }
+     *
+     * export class B extends A {
+     *   a: number;
+     *   b: string;
+     * }
+     *
+     * export class C extends B {
+     *   c: boolean;
+     * }
+     * ```
+     *
+     * Calling `findMembersWithInheritance` on `C` will return `B.a`, `B.b`, and `C.c`. Calling the
+     * method on `B` will return `B.a` and `B.b`. And calling the method on `A` will return just
+     * `A.a`.
+     *
+     * The inherited members returned by this method may be incomplete. If so, a flag on the result
+     * object will indicate this, and the provided callback will be invoked with more specific
+     * messages explaining the errors. Some scenarios include:
+     *
+     * - Interface extending from a type alias.
+     * - Class extending from a variable.
+     * - Extending from a declaration not present in the model (e.g. external package).
+     * - Extending from an unexported declaration (e.g. ae-forgotten-export). Common in mixin
+     *   patterns.
+     * - Unexpected runtime errors...
+     *
+     * Lastly, be aware that the types of inherited members are returned with respect to their
+     * defining class as opposed to with respect to the inheriting class. For example, consider
+     * the following:
+     *
+     * ```
+     * export class A<T> {
+     *   a: T;
+     * }
+     *
+     * export class B extends A<number> {}
+     * ```
+     *
+     * When called on `B`, this method will return `B.a` with type `T` as opposed to type
+     * `number`, although the latter is more accurate.
+     */
+    public findMembersWithInheritance(
+      messageCallback?: FindMembersWithInheritanceMessageCallback
+    ): IFindMembersWithInheritanceResult {
+      // This method uses the same logic as that in `ApiItemContainerMixin` to store members by
+      // by name and kind.
+      const membersByName: Map<string, ApiItem[]> = new Map();
+      const membersByKind: Map<ApiItemKind, ApiItem[]> = new Map();
+
+      const toVisit: ApiItem[] = [];
+      let next: ApiItem | undefined = this;
+      let maybeIncompleteMembers: boolean = false;
+
+      while (next) {
+        const membersToAdd: ApiItem[] = [];
+
+        // For each member, check to see if we've already seen a member with the same name
+        // previously in the inheritance tree. If so, we know we won't inherit it, and thus
+        // do not add it to our `membersToAdd` array.
```

Comment:
```ts
interface A {
  /** This is A.X */
  x: string;
}
interface B {
  /** This is B.X */
  x: string;
}
interface C extends A, B { }
```
Which is reported as the "inherited" `x` for `C`?  `A.x` or `B.x`?

Reply:
From our short Zulip conversation: You said:

> It probably doesn't matter too much for our case, but it would be nice if the selection was deterministic (ideally alphabetical instead of based on declaration order). And if it prefers the item with more documentation (for example if `A.x` has 3 paragraphs of text whereas `B.x` has only a few words). Not necessary to implement in this PR, though.

I updated the implementation to be deterministic based upon alphabetical order. I didn't add any logic yet to prefer the item with more documentation (I thought this was a good idea but thought it made more sense in a subsequent PR).